src/libraries/System.Private.CoreLib/src/System/Int128.cs

```diff
@@ -1043,57 +1043,29 @@ public static Int128 Log2(Int128 value)
         /// <inheritdoc cref="IComparisonOperators{TSelf, TOther, TResult}.op_LessThan(TSelf, TOther)" />
         public static bool operator <(Int128 left, Int128 right)
         {
-            if (IsNegative(left) == IsNegative(right))
-            {
-                return (left._upper < right._upper)
-                    || ((left._upper == right._upper) && (left._lower < right._lower));
-            }
-            else
-            {
-                return IsNegative(left);
-            }
+            return ((long)left._upper < (long)right._upper)
```

Comment:
A comment (in the code) explaining why this is valid would be beneficial.

Actually brute forcing this for some `struct S16 { byte _lower; byte _upper; }` is pretty trivial to do, but so is describing why this works out given the two's complement representation. -- Notably this comes down to the fact that all negative values have the sign bit set, so if `lhs.upper < rhs.upper` in signed form, then either lhs/rhs have different signs, and the single comparison was sufficient or they have the same sign and the same was true.

------

There may be other optimizations available here as well... In particular, comparisons `x cmp y` in general are done as if `x - y` and then setting CPU flags, without actually producing the result. That general premise is why the "most optimal" pattern is actually `cmp lower; sbb upper; setcc`

Reply:
I tried my best documenting it with comments.